packages/enzyme/src/ShallowWrapper.js

```diff
@@ -180,6 +185,55 @@ function isPureComponent(instance) {
   return instance && instance.isPureReactComponent;
 }
 
+function getChildContext(node, hierarchy, renderer) {
+  const { instance, type: Component } = node;
+  if (!Component.childContextTypes) {
+    return {};
+  }
+  const childContext = instance.getChildContext();
+  Object.keys(childContext).forEach((key) => {
+    if (!(key in Component.childContextTypes)) {
```

Comment:
are we using `in` here because React does? it seems like this should be an own property check.

Reply:
Yeah I tried to match react as much as possible [and they are using `in`](https://github.com/facebook/react/blob/1454a8be03794f5e0b23a7e7696cbbbdcf8b0f5d/packages/react-dom/src/server/ReactPartialRenderer.js#L633).

Reply:
Let's add a comment then, above line 193, linking to https://github.com/facebook/react/blob/1454a8be03794f5e0b23a7e7696cbbbdcf8b0f5d/packages/react-dom/src/server/ReactPartialRenderer.js#L630-L637